### ISSUE TITLE
ULP mismatch in Exp

### ISSUE THREAD

**Issue by karthikkattaopenxpu** [tenstorrent/tt-metal#24351]
### Describe the bug

 **outputs_ref**
tensor([[[1.9609]],

        [[1.2031]],

        [[1.5781]],

        [[2.1094]],

        [[1.2969]],

        [[2.6719]],

        [[2.1406]],

        [[1.4922]],

        [[1.0391]],

        [[2.2812]],

        [[1.0859]],

        [[2.5781]]], dtype=torch.bfloat16)

**outputs_dev**
tensor([[[1.9531]],

        [[1.1797]],

        [[1.5625]],

        [[2.1094]],

        [[1.2812]],

        [[2.6406]],

        [[2.1250]],

        [[1.4766]],

        [[1.0234]],

        [[2.2656]],

        [[1.0703]],

        [[2.5469]]], dtype=torch.bfloat16, device='cuda:0')

Can this mis-match be fixed ? Please provide your comments.

### Steps to reproduce the issue

```
def aten__exp__default_test_0000(dtype_default = torch.bfloat16, device = "cuda:0", seed = 404, compute_reference = True):
    if seed is not None:
        torch.manual_seed(seed)
    self = torch.rand([12,1,1], dtype=dtype_default, device="cpu")
    outputs_ref = None
    if compute_reference:
        outputs_ref = torch.ops.aten.exp.default(self)
    self = self.to(device)
    outputs_dev = torch.ops.aten.exp.default(self)
    err_limits_ulp_max_mae_mse = (None, None, None, None)
    import pdb
    pdb.set_trace()
    return outputs_dev, outputs_ref, err_limits_ulp_max_mae_mse, "aten.exp.default", "0000", None
```

**Comment by KalaivaniMCW** [comment]
Hi @karthikkattaopenxpu 
Could you please share the corresponding input values for these ? 
We could check this against the alternate version of `exp` that is in progress.

If it is always seed = 404, for the tensor generation, then we can extract that.

**Comment by KalaivaniMCW** [comment]
@VirdhatchaniKN  

Could you add this test to the exp_accurate branch you have 
```
    torch.manual_seed(seed=404)
    self = torch.rand([12,1,1],) // bf16
```

**Comment by karthikkattaopenxpu** [comment]
Please find the input values here.

[log.txt](https://github.com/user-attachments/files/21105737/log.txt)

I tried both ttnn.exp2 and ttnn.expm1 .. both exhibited greater errors than ttnn.exp.

**Comment by KalaivaniMCW** [comment]
Tested the input with the alternate version of `exp` and these are the results:

input_tensor
`torch.tensor([[[0.6719]], [[0.1836]], [[0.4570]], [[0.7500]], [[0.2617]], [[0.9805]], [[0.7617]], [[0.4023]], [[0.0352]], [[0.8242]], [[0.0820]], [[0.9453]]], dtype=torch.bfloat16)`

tt_res tensor
`([[[1.9531]], [[1.1953]], [[1.5781]], [[2.1094]], [[1.2891]], [[2.6562]], [[2.1250]], [[1.4922]], [[1.0312]], [[2.2656]], [[1.0781]], [[2.5625]]], dtype=torch.bfloat16) `

torch_res tensor
`([[[1.9609]], [[1.2031]], [[1.5781]], [[2.1094]], [[1.2969]], [[2.6719]], [[2.1406]], [[1.4922]], [[1.0391]], [[2.2812]], [[1.0859]], [[2.5781]]], dtype=torch.bfloat16) `

abs_diff tensor
`([[[0.0078]], [[0.0078]], [[0.0000]], [[0.0000]], [[0.0078]], [[0.0156]], [[0.0156]], [[0.0000]], [[0.0078]], [[0.0156]], [[0.0078]], [[0.0156]]], dtype=torch.bfloat16)
`
<img width="1512" height="982" alt="Image" src="https://github.com/user-attachments/assets/4ddef9f0-effd-4c72-bc8c-3489288bbea0" />
<img width="1512" height="982" alt="Image" src="https://github.com/user-attachments/assets/42977268-f84a-4b80-9280-d3405f03fefd" />

**Comment by karthikkattaopenxpu** [comment]
@KalaivaniMCW  What is the alternative version of exp api ? Is it pending for merge ? Please suggest.

**Comment by KalaivaniMCW** [comment]
@karthikkattaopenxpu Yes, it is pending for merge. We're not yet ready to replace the existing `exp` op with the alternate version due to perf constraints. Do you have a timeline on when you need this ?

**Comment by VirdhatchaniKN** [comment]
Update on Exp : [Link](https://github.com/tenstorrent/tt-metal/issues/24306#issuecomment-3102788705)

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/pull/25946
merged to main
